azurerm/internal/services/media/media_asset_resource.go

```diff
@@ -0,0 +1,178 @@
+				),
+			},
+
+			"alternate_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"container": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"storage_account_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmMediaAssetsCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Media.AssetsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	assetName := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	accountName := d.Get("media_services_account_name").(string)
+	description := d.Get("description").(string)
+	alternateId := d.Get("alternate_id").(string)
+	container := d.Get("container").(string)
+	storageAccountName := d.Get("storage_account_name").(string)
+
+	parameters := media.Asset{
+		AssetProperties: &media.AssetProperties{
+			Description:        &description,
+			AlternateID:        &alternateId,
+			StorageAccountName: &storageAccountName,
+		},
+	}
+
+	if container != "" {
+		parameters.Container = &container
+	}
+
+	if _, e := client.CreateOrUpdate(ctx, resourceGroup, accountName, assetName, parameters); e != nil {
+		return fmt.Errorf("Error creating Asset %q in Media Services Account %q (Resource Group %q): %+v", assetName, accountName, resourceGroup, e)
+	}
+
+	asset, err := client.Get(ctx, resourceGroup, accountName, assetName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Asset %q in Media Services Account %q (Resource Group %q): %+v", assetName, accountName, resourceGroup, err)
+	}
+	log.Printf(*asset.ID)
+	d.SetId(*asset.ID)
+
+	return resourceArmMediaAssetsRead(d, meta)
+}
+
+func resourceArmMediaAssetsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Media.AssetsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.MediaAssetsID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.AccountName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Asset %q was not found in Media Services Account %q and Resource Group %q - removing from state", id.Name, id.AccountName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Asset %q from Media Services Account %q (Resource Group %q): %+v", id.Name, id.AccountName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("media_services_account_name", id.AccountName)
+	d.Set("description", resp.AssetProperties.Description)
+	d.Set("alternate_id", resp.AssetProperties.AlternateID)
```

Comment:
could we put this inside an if statement checking that AssetPropertiesis not nil?